modules/library/main/src/main/java/org/geotools/styling/visitor/UomRescaleStyleVisitor.java

```diff
@@ -197,6 +197,7 @@ public void visit(PolygonSymbolizer poly) {
         Unit<Length> uom = copy.getUnitOfMeasure();
         rescaleStroke(copy.getStroke(), uom);
         rescaleFill(copy.getFill(), uom);
+        scaleIntArrayOption(copy.getOptions(), TextSymbolizer.GRAPHIC_MARGIN_KEY, uom);
```

Comment:
Ouch ... I added the PolygonSymbolizer constant to make it clear and forgot to updated it (facepalm).